apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.dto.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  ArrayMaxSize,
+  IsArray,
+  IsBoolean,
+  IsEnum,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  Length,
+  ValidateNested,
+} from 'class-validator';
+import { Type } from 'class-transformer';
+import { MAX_NAME_LENGTH } from '@novu/application-generic';
+import { WorkflowCreationSourceEnum } from '@novu/shared';
+import { UpsertStepData } from './upsert-step.data';
+import { PreferencesRequestUpsertData } from './preferences-request-upsert.data';
+
+export class UpsertWorkflowDto {
```

Comment:
The usecase should not care that this is a DTO, this should be a Command.

```suggestion
export class UpsertWorkflowCommand {
```